Source/WebCore/html/HTMLVideoElement.cpp

```diff
@@ -675,8 +679,12 @@ void HTMLVideoElement::serviceRequestVideoFrameCallbacks(ReducedResolutionSecond
 void HTMLVideoElement::mediaPlayerEngineUpdated()
 {
     HTMLMediaElement::mediaPlayerEngineUpdated();
-    if (!m_videoFrameRequests.isEmpty() && player())
+    if (!m_videoFrameRequests.isEmpty() && player()) {
+#if PLATFORM(COCOA) && !HAVE(AVSAMPLEBUFFERDISPLAYLAYER_COPYDISPLAYEDPIXELBUFFER)
```

Comment:
Why not calling willBeAskedToPaintGL() in the startVideoFrameMetadataGathering() specialisation?

Reply:
Will there be a scenario where `startVideoFrameMetadataGathering()` is called without the need for painting?

Reply:
Some websites compute frame rates from rvfc for logging/qa metrics purposes.

Reply:
In that case, this change will break those websites as @whsieh mentioned since the display layer will be destroyed on rvfcv for older builds.